section3/reflection.md

```diff
@@ -1,13 +1,32 @@
 ## Section 3 Reflection
 
-1. In Section 3, we removed even more supports and pushed you to learn a bit more independently, read through technical documentation, and use your resources more. How did that feel?
+1. In Section 3, we removed even more supports and pushed you to learn a bit more independently, read through technical documentation, and use your resources more. How did that feel?  
 
-2. Think of a web or mobile application you use regularly. Where is an example of a place that the developers probably use Arrays? Explain.
+   It felt great, because I was able to connect the dots about what the code was actually
+   doing.
 
-3. In your own words, explain what an Array method is.
+2. Think of a web or mobile application you use regularly. Where is an example of a place that the developers probably use Arrays? Explain.  
 
-4. What is the purpose of a loop? When we would use a loop in conjunction with array vs. without?
+   I play Spades Plus often, I can see the developers using arrays to store
+   decks of cards.
 
-5. If you had to teach a 5 year old about loops, how would you explain it?
+3. In your own words, explain what an Array method is.  
 
-6. What questions do you still have about Arrays and loops?
+   An array method is a way to access and or update values of an array.
+
+4. What is the purpose of a loop? When we would use a loop in conjunction with array vs. without?  
+
+   The purpose of loops, are to help developers write code in their programs that
+    will run repetitive tasks. A loop would be used in conjunction with an array
+    when you want to run the same code for each array item. A loop wouldn't be used
+    with an array, if you were running a calculation on one variable .
+
+5. If you had to teach a 5 year old about loops, how would you explain it?  
+
+   Loop are used to tell your computer program to run the same task for a set  
+   number of times.
+
+6. What questions do you still have about Arrays and loops?  
+
+   While doing some digging through the MDN docs, I ran across the `forEach()` method.  
```

Comment:
Cool cool, yes, you'll get into `forEach` in Mod 1 and many more in Mod 2! "Under the hood" as they say, `forEach` is built with a for loop. Most devs prefer the syntax of a `forEach` so use it instead. There are SOME cases where `forEach` won't do the job for one reason or another, and at Turing we want students to deeply understand the fundamentals so we have you learn and use for loops for a while first, then introduce the shiny methods.